ui/src/app/workflows/components/workflow-details/artifact-panel.tsx

```diff
@@ -53,6 +53,13 @@ export const ArtifactPanel = ({
     }, [downloadUrl]);
     useCollectEvent('openedArtifactPanel');
 
+    const spec = execSpec(workflow);
+    const artifacts = spec.templates.find(t => t.name === workflow.status.nodes[artifact.nodeId].templateName)?.outputs?.artifacts;
+    let artifactGCStrategy = '';
```

Comment:
I wonder if we need to do error handling for artifact.nodeId not being found in workflow.status.nodes (which I recognize probably shouldn't happen unless there's a bug somewhere). @alexec Do you have an opinion? Maybe the result is an error in the console in either case?